### ISSUE TITLE
Does this image supports xDebug?

### ISSUE THREAD

**Issue by KIVagant** [bitnami/bitnami-docker-wordpress#116]
Hello. Is it possible to use this image with xDebug?
The question is related to the use-case described in https://github.com/bitnami/bitnami-docker-wordpress/issues/115

**Comment by tompizmor** [comment]
Hello @KIVagant,

Do you mind check [this comment](https://github.com/bitnami/bitnami-docker-php-fpm/issues/100#issuecomment-363141887) about how to add xdebug extension to the bitnami/php-fpm container image? 
The instructions should work for the WordPress image too.

**Comment by KIVagant** [comment]
Thank you, @tompizmor 

I would like to create a Dockerfile.xdebug with everything inside from the comment you mentioned and left it somewhere here for history. I don't know about the ETA because I've solved the problem for me with a different image, so if anyone decide to close the ticket – it's okay, that comment is enough for start.

**Comment by javsalgar** [comment]
Hi! We would really appreciate if you shared with us the content when you have it ready :) Despite it is closed, do not hesitate to write

**Comment by rquadling** [comment]
https://gist.github.com/rquadling/53fd99994d653b0a1b88f4a0705ae655

**Comment by KIVagant** [comment]
@rquadling , great, thank you!

**Comment by rquadling** [comment]
Thank you for the feedback. One point in return. The remote_port being 9000. I choose 9001 as I am running nginx in a separate container and that is on 9000. I couldn't find any consensus on which way to go. I think either way, both are right if configured correctly.

**Comment by CharlyRipp** [comment]
Hi there - was hoping one of you could shed some light on what I'm doing wrong as this pretty close to what I'm doing, before I take it to stack overflow.

BACKGROUND - I deploy to kubernetes and prefer to keep the docker image 'pure', so during local development I run a docker-compose and just mount over the `/bitnami/wordpress` folder with a development export and configuration changes to run against a local database (also a dev export). All runs fine -- looking at adding xdebug ability.

SETUP - Currently I've created a mimic of the `/app-entrypoint.sh` and added commands to install xdebug before the startup -

    #!/bin/bash -e
    
    . /opt/bitnami/base/functions
    . /opt/bitnami/base/helpers
    
    print_welcome_page

    #### Also tried moving the install up here ####
    
    if [[ "$1" == "nami" && "$2" == "start" ]] || [[ "$1" == "httpd" ]]; then
      . /apache-init.sh
      . /wordpress-init.sh
      nami_initialize apache php mysql-client wordpress
      info "Starting wordpress... "
    fi
    
    #### Difference begin ####
    apt-get update
    install_packages php-pear autoconf build-essential
    pecl install xdebug
    cat << EOF >> /opt/bitnami/php/lib/php.ini
    zend_extension=/opt/bitnami/php/lib/php/extensions/xdebug.so
    xdebug.remote_enable=1
    xdebug.remote_autostart=1
    xdebug.remote_connect_back=0
    xdebug.remote_host=docker.for.mac.localhost
    xdebug.remote_port=9000
    xdebug.remote_key=PHPSTORM
    EOF
    #### Difference end ####
    
    exec tini -- "$@"

Seems to startup fine - no excess logs outside the normal apt-get and pecl output.

Using the Xdebugger-helper extension in chrome, it seems to see it's a valid debug point.

Over in PHPStorm, I've tried with and without a server configured. When configured, it's set to localhost:80 (docker image is exposed on 80, but hosts file entries are used to mimic multi-site domains with `.com` replaced as `.local`)
Again, tried the PHP Remote Debug with and without the Filter connection by IDE key, set to "PHPSTORM".

Made sure PHPStorm was listening on 9000

    COMMAND   NODE            NAME
    phpstorm  TCP     *:cslistener (LISTEN)

I can't seem to get the connection established to PHPStorm -- any rough ideas to continue down?

**Comment by juan131** [comment]
Hi @CharlyRipp 

You're installing XDebug (and enabling it) in the **entrypoint.sh** which is highly discouraged. Even more, when you're deploying this image on K8s.

Please note that in K8s, a container (a pod to be more correct) can be restarted in any moment, so your images' initialisation must be as quick as possible so your app's downtime is minimal. Installing XDebug during the initialisation is going to affect significantly WordPress's boot time.

Instead of doing so, you should be creating your own image (moving all these steps to "build time"):

```Dockerfile
FROM bitnami/wordpress

RUN install_packages php-pear autoconf build-essential
RUN pecl install xdebug
RUN ...
```

Regarding the issue with PHPStorm, how are you accessing it? Are you using a "port-forward" to access your Pods? The port 9000 is not opened by default..

**Comment by CharlyRipp** [comment]
Hey @juan131 -- apologies for the confusion, I only run this process locally. The k8s contains stays the pure image. Trying to keep it slim as we scale (currently running 7 instances minimum)

Anyways, I hadn't tried port forwarding -- figured the `9000` port would have communicated container -> host via `docker.for.mac.localhost`. Also the 'validation' worked via PHPStorm on that.

With what you said, I attempted to run it as network_host, but for some reason couldn't break this image from trying to connect to the database via container name, which can't be done when on host.

Ended up reverting that and switched PHPStorm to listen on 9001 and the `xdebug.remote_port=9001` -- which worked....
Nothing else is on 9000 (`sudo lsof -i :9000`), so no idea what's different now.

Thank you for your help leading me to trying that. I'm gonna attempt to see why 9000 wasn't working to know for sure - would rather have 0-config on developers machines.

**Comment by juan131** [comment]
Hi @CharlyRipp 

> With what you said, I attempted to run it as network_host, but for some reason couldn't break this image from trying to connect to the database via container name, which can't be done when on host.

I didn't mean you to change the network configuration, I just though you were running this custom container on K8s.

> Ended up reverting that and switched PHPStorm to listen on 9001 and the xdebug.remote_port=9001 -- which worked....
> Nothing else is on 9000 (`sudo lsof -i :9000`), so no idea what's different now.

That's very weird... That shouldn't be a problem at all.

> Thank you for your help leading me to trying that. I'm gonna attempt to see why 9000 wasn't working to know for sure

You're welcome! Just for curiosity, please let us know if you find the reason why it's not working in the port 9000

**Comment by CharlyRipp** [comment]
Nothing much more to add, but I've exhausted everything I could think of to figure out the port 9000 issue. Even tried on a fresh installation of OSX 10.14 - just happened to be re-upping the other day and figured I'd give it a wack with only PHPStorm and Docker installed for the most part.

I did end up switching to using `host.docker.internal` and testing on Windows 10, where port 9000 worked fine. In all my research, can't find anything special about 9000 and OSX/Docker.
